GaelAlburo/backend-project
# Add Products API with Full Functionality and Containerization Support

**Description:**  
This PR introduces the fully functional Products API, along with the necessary configuration for deploying it as a containerized service. It includes all core functionality for managing products, a `Dockerfile` for building the container image, and a `requirements.txt` file for managing dependencies. Additionally, `.gitignore` has been updated to exclude unnecessary files generated during development.

**Changes:**  
 - Added a complete implementation of the Products API, enabling CRUD operations for products and providing all required routes for product management.  
- Created a `Dockerfile` for containerizing the Products API.  
- Added a `requirements.txt` file to specify dependencies required for the API.  
- Updated `.gitignore` to exclude `__pycache__` directories generated within the `products_API` directory during development.  

**Impact:**  
- Adds the Products API as a critical backend component, enhancing the project with product management functionality.  
- Enables containerization of the Products API for scalable deployment in Docker environments.  

**Testing:**  
1. Build the Docker image for the Products API and confirm successful creation.  
2. Run the containerized Products API and verify that all routes (e.g., CRUD operations) function as expected.
